From a0652e9e68ed017177644900bda449865dbc016f Mon Sep 17 00:00:00 2001
From: Jens Rehsack <sno@netbsd.org>
Date: Fri, 12 Sep 2014 11:00:41 +0200
Subject: [PATCH 4/4] don't printout debug message when DBG is off

even _drv_always debug messages are omitted - respect silence configuration
---
 drivers/net/wireless/rtl8189es/include/rtw_debug.h | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/rtl8189es/include/rtw_debug.h b/drivers/net/wireless/rtl8189es/include/rtw_debug.h
index 67edf7f..5e5ee19 100644
--- a/drivers/net/wireless/rtl8189es/include/rtw_debug.h
+++ b/drivers/net/wireless/rtl8189es/include/rtw_debug.h
@@ -161,21 +161,30 @@ extern void rtl871x_cedbg(const char *fmt, ...);
 	#define MSG_8192C do {} while(0)
 	#define DBG_8192C do {} while(0)
 	#define DBG_871X_LEVEL do {} while(0)
+	#define _DBG_871X_LEVEL do {} while(0)
+	#define DBG_871X_SEL do {} while(0)
+	#define DBG_871X_SEL_NL do {} while(0)
 #else
 	#define DBG_871X(x, ...) do {} while(0)
 	#define MSG_8192C(x, ...) do {} while(0)
 	#define DBG_8192C(x,...) do {} while(0)
 	#define DBG_871X_LEVEL(x,...) do {} while(0)
+	#define _DBG_871X_LEVEL(x, ...) do {} while(0)
+	#define DBG_871X_SEL(x, ...) do {} while(0)
+	#define DBG_871X_SEL_NL(x, ...) do {} while(0)
 #endif
 
 #undef _dbgdump
 #undef _seqdump
 
+#define RTW_DBGDUMP 0 /* 'stream' for _dbgdump */
+
 #ifndef _RTL871X_DEBUG_C_
 	extern u32 GlobalDebugLevel;
 	extern u64 GlobalDebugComponents;
 #endif
 
+#if defined(DBG) && (DBG != 0)
 #if defined(PLATFORM_WINDOWS) && defined(PLATFORM_OS_XP)
 	#define _dbgdump DbgPrint
 	#define _seqdump(sel, fmt, arg...) _dbgdump(fmt, ##arg)
@@ -189,6 +198,7 @@ extern void rtl871x_cedbg(const char *fmt, ...);
 	#define _dbgdump printf
 	#define _seqdump(sel, fmt, arg...) _dbgdump(fmt, ##arg)
 #endif
+#endif
 
 #define DRIVER_PREFIX "RTL871X: "
 
@@ -221,9 +231,9 @@ extern void rtl871x_cedbg(const char *fmt, ...);
 	}while(0)
 
 #if defined(_seqdump)
-#define RTW_DBGDUMP 0 /* 'stream' for _dbgdump */
 
 /* dump message to selected 'stream' */
+#undef DBG_871X_SEL
 #define DBG_871X_SEL(sel, fmt, arg...) \
 	do {\
 		if (sel == RTW_DBGDUMP)\
@@ -234,6 +244,7 @@ extern void rtl871x_cedbg(const char *fmt, ...);
 	}while(0)
 
 /* dump message to selected 'stream' with driver-defined prefix */
+#undef DBG_871X_SEL
 #define DBG_871X_SEL_NL(sel, fmt, arg...) \
 	do {\
 		if (sel == RTW_DBGDUMP)\
@@ -245,6 +256,10 @@ extern void rtl871x_cedbg(const char *fmt, ...);
 
 #endif /* defined(_seqdump) */
 
+#else /* defined(_dbgdump) */
+
+#define DBG_871X_EXP(level, EXP) do { (void)(level); (void)(EXP); } while (0)
+
 #endif /* defined(_dbgdump) */
 
 #ifdef CONFIG_DEBUG
-- 
1.9.1

